### PR TITLE
Use 404 when unit or node could not be found

### PR DESCRIPTION
**Pull request by janisz**

### CODE REVIEW

File: api/handlers.go [hunk 1]
```diff
@@ -52,9 +52,14 @@ func (h *handler) getAllUnitsHandler(w http.ResponseWriter, _ *http.Request) {
 // /api/v1/system/health/units/:unit_id:
 func (h *handler) getUnitByIDHandler(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
-	unitResponse, err := h.monitoringResponse.GetUnit(vars["unitid"])
+	unitId := vars["unitid"]
```
Comment (gracedo) [inline api/handlers.go]: just wondering why this was pulled out into a var (doesn't look to be used elsewhere)

File: api/handlers.go [hunk 2]
```diff
@@ -78,10 +88,15 @@ func (h *handler) getNodesByUnitIDHandler(w http.ResponseWriter, r *http.Request
 // /api/v1/system/health/units/:unit_id:/nodes/:node_id:
 func (h *handler) getNodeByUnitIDNodeIDHandler(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
-	nodePerUnit, err := h.monitoringResponse.GetSpecificNodeForUnit(vars["unitid"], vars["nodeid"])
+	unitId := vars["unitid"]
+	nodePerUnit, err := h.monitoringResponse.GetSpecificNodeForUnit(unitId, vars["nodeid"])
```
Comment (gracedo) [inline api/handlers.go]: it's a little inconsistent that `unitId` is pulled out into a var but not `vars["nodeid"]` - I think maybe it was ok to leave it as originally written?

Comment (janisz) [reply api/handlers.go]: Fixed

File: api/handlers.go (hunk 1 above)
Comment (janisz) [reply api/handlers.go]: Accidentally committed my WIP version. Thanks for pointing out.

**Pull request merged**